AIO/src/org/aio/gui/Gui.java

```diff
@@ -25,7 +25,7 @@
 
     private boolean started;
     private JPanel taskList = new JPanel();
-    private List<TaskPanel> taskPanels = new ArrayList<>();
+    private TreeMap<String, TaskPanelContent> taskPanels = new TreeMap<>();
```

Comment:
I think we should be using a `LinkedHashMap` here rather than a `TreeMap`. A `TreeMap` preserves ordering based on the key, whereas a `LinkedHashMap` preserves the insertion order.

Reply:
Ah that's what I was looking for, thanks. I'll change that over.